src/main/java/com/android/tools/build/bundletool/io/ApkSerializerManager.java

```diff
@@ -298,11 +291,19 @@ private ModuleSplit modifyApk(ModuleSplit moduleSplit, int variantNumber) {
             .setApkTargeting(moduleSplit.getApkTargeting())
             .build();
 
-    return moduleSplit
-        .toBuilder()
-        .setAndroidManifest(
-            apkModifier.modifyManifest(moduleSplit.getAndroidManifest(), apkDescription))
-        .build();
+    ModuleSplit.Builder builder = moduleSplit.toBuilder();
+
+    if (moduleSplit.isMasterSplit() && debuggable.isPresent()) {
+
+        builder.setAndroidManifest(
+              apkModifier.modifyManifest(
```

Comment:
We used to call `apkModifier.modifyManifest` on all APKs but now you've changed it to call it only on the master split.